File: compare_logos.py

import re
import os

def get_logo_src(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    # Find the img tag with alt="Smart Stock Logo" and extract src
    match = re.search(r'<img[^>]+alt="Smart Stock Logo"[^>]+src="([^"]+)"', content)
    if match:
        return match.group(1)
    return None

login_path = r'c:\Users\Vini\Documents\Projeto Smart Stock\Projeto-Smart-Stock\login.html'
signup_path = r'c:\Users\Vini\Documents\Projeto Smart Stock\Projeto-Smart-Stock\signup.html'

login_src = get_logo_src(login_path)
signup_src = get_logo_src(signup_path)

if login_src == signup_src:
    print("MATCH: The logo src strings are identical.")
else:
    print("MISMATCH: The logo src strings are different.")
    if login_src and signup_src:
        print(f"Login length: {len(login_src)}")
        print(f"Signup length: {len(signup_src)}")
        # Print a small part of the difference if any
        for i in range(min(len(login_src), len(signup_src))):
            if login_src[i] != signup_src[i]:
                print(f"First mismatch at index {i}: '{login_src[i]}' vs '{signup_src[i]}'")
                break
    else:
        print(f"Login found: {login_src is not None}")
        print(f"Signup found: {signup_src is not None}")

File: compare_logos.js

const fs = require('fs');
const path = require('path');

function getLogoSrc(filePath) {
    const content = fs.readFileSync(filePath, 'utf8');
    const match = content.match(/<img[^>]+alt="Smart Stock Logo"[^>]+src="([^"]+)"/);
    return match ? match[1] : null;
}

const loginPath = 'c:\\Users\\Vini\\Documents\\Projeto Smart Stock\\Projeto-Smart-Stock\\login.html';
const signupPath = 'c:\\Users\\Vini\\Documents\\Projeto Smart Stock\\Projeto-Smart-Stock\\signup.html';

const loginSrc = getLogoSrc(loginPath);
const signupSrc = getLogoSrc(signupPath);

if (loginSrc === signupSrc) {
    console.log("MATCH: The logo src strings are identical.");
} else {
    console.log("MISMATCH: The logo src strings are different.");
    if (loginSrc && signupSrc) {
        console.log(`Login length: ${loginSrc.length}`);
        console.log(`Signup length: ${signupSrc.length}`);
        const minLen = Math.min(loginSrc.length, signupSrc.length);
        for (let i = 0; i < minLen; i++) {
            if (loginSrc[i] !== signupSrc[i]) {
                console.log(`First mismatch at index ${i}: '${loginSrc[i]}' vs '${signupSrc[i]}'`);
                break;
            }
        }
    } else {
        console.log(`Login found: ${!!loginSrc}`);
        console.log(`Signup found: ${!!signupSrc}`);
    }
}
